flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitor.java

```diff
@@ -276,4 +294,18 @@ public int getServerPort() {
 	private static RuntimeMonitorHandler handler(RequestHandler handler) {
 		return new RuntimeMonitorHandler(handler);
 	}
+
+	private static String getFilePath(String pattern) {
```

Comment:
There is a  list files version which takes a filter. You can directly use that one.